File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int brack = 0, depth_bracket = 0;
        for(char bracket: s){
            if(bracket == '(')brack++;
            else if(bracket ==')') brack--;
            depth_bracket = max(depth_bracket, brack);
        }
        return depth_bracket;
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.go
func maxDepth(s string) int {
	max := 0
    count := 0
	for _, ch := range s {
		if ch == '(' {
			count++
			if count > max {
				max = count
			}
		} else if ch == ')' {
			count--       
		}
	}
	return max
}